src/meta/processors/admin/BalanceTask.cpp

```diff
@@ -77,58 +90,82 @@ void BalanceTask::invoke() {
         case Status::CATCH_UP_DATA: {
             LOG(INFO) << taskIdStr_ << "Waiting for the data catch up.";
             SAVE_STATE();
-            client_->waitingForCatchUpData(spaceId_, partId_).thenValue([this](auto&& resp) {
+            client_->waitingForCatchUpData(spaceId_, partId_, dst_).thenValue([this](auto&& resp) {
                 if (!resp.ok()) {
                     ret_ = Result::FAILED;
```

Comment:
It could take quite a long time to catch up data, maybe we can't return `FAILED` here

Reply:
We should give a response on storage layer.